src/docs/arch-rts.dox

```diff
@@ -1,113 +1,76 @@
-conditions.
-1. Table is modified
-2. The checkpoint durable start/stop timestamp is greater than the rollback timestamp.
-3. There is no durable timestamp in any checkpoint.
-4. Has prepared updates
-5. Has updates from transactions greater than checkpoint snapshot (only in restart phase)
-
-There are some special conditions where the table is skipped for rollback to stable.
-1. Empty file
-2. Table has timestamp updates but there is no stable timestamp set.
-3. Files that don't exist
-4. Files that are corrupted.
-
-If the table has no timestamp updates, then the history store table is
-scanned to remove any historical versions related to this table as these
-older versions no longer required.
-
-Once all the tables are process for rollback to stable, at the end the history store
-is processed to remove any unstable updates more than the stable timestamp.
-
-@section rts-how How rollback to stable fixes the unstable updates
-
-Once a table is identified to perform rollback to stable, it reads the pages into
-the cache if they don't exist and process it to remove the unstable updates.
-
-There are two types of rollbacks that rollback to stable performs:
-1. Rolling back unstable fast truncate
-2. Rolling back unstable updates
-
-All internal pages are traversed to rollback unstable fast truncate operations and
-leaf pages are traversed to remove the unstable updates.
+For each selected table, its pages are read into the cache. The unstable in-memory updates and the
+unstable historical versions from the history store are removed. The on-disk version is replaced by
+the latest stable in-memory update if any, otherwise by the latest stable version from the history
+store. If there is no stable in-memory updates and no stable version in the history store, the data
+store entry is completely removed. In the situation where the table has no timestamped updates, all
+the historical versions related to this table are removed from the history store as they are no
+longer required and the on-disk value is preserved.
 
-Once the leaf page is identified to rollback the updates, it is performed in the
-following order.
-1. Check smallest insert lists on the page
-2. Traverse through all the on-disk keys
-    a. Check update list
-    b. Check insert list
-    c. Check the on-disk version if no stable update found in the update list.
-3. Traverse through the reconciled pages to abort any history store updates.
+Finally, RTS is triggered on startup, shutdown or when the application is initiated. It is worth
+noting that it cannot run if there is any transaction activity happening in WiredTiger.
 
-@section rts-abort-update How rollback to stable aborts in-memory updates
+@section rts-pages Internal and leaf pages
 
-Traverse through all the updates in the update list and abort them until a stable
-update is found.
+On internal pages, RTS can roll back unstable fast truncate operations.
```

Comment:
This has changed now with the fast truncate information written to the disk. There is no special logic used to remove fast truncate from the internal pages. It works similarly to other unstable updates. The fast truncated page gets read into the cache and removes all the tombstones that are added.

Reply:
Does it mean we don't need to specifically mention `fast-truncate`?

Reply:
Yes, that's correct. No need to mention internal and leaf page processing separately.

Reply:
There are a couple places that explicitly check the delete information, specifically in the page skip-or-not checks. This is probably adequately explained in the FT docs though.

Reply:
May be explaining in another section how RTS decides a page to be processed for checking any unstable updates with all the conditions.